diff --git a/keywords/_formatLimit.js b/keywords/_formatLimit.js
--- a/keywords/_formatLimit.js
+++ b/keywords/_formatLimit.js
@@ -58,7 +58,8 @@
 
 
 function extendFormats(ajv) {
-  var formats = ajv._formats;
+  var formats = ajv._formats || ajv.formats;
+  if (!formats) return;
   for (var name in COMPARE_FORMATS) {
     var format = formats[name];
     // the last condition is needed if it's RegExp from another window
